llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -13196,6 +13207,15 @@ Value *BoUpSLP::vectorizeTree(TreeEntry *E, bool PostponedPHIs) {
           False = Builder.CreateIntCast(False, VecTy, GetOperandSignedness(2));
       }
 
+      unsigned CondNumElements = getNumElements(Cond->getType());
+      unsigned TrueNumElements = getNumElements(True->getType());
+      if (CondNumElements != TrueNumElements) {
+        // When the return type is i1 but the source is fixed vector type, we
+        // need to duplicate the condition value.
+        Cond = Builder.CreateShuffleVector(
+            Cond, createReplicatedMask(TrueNumElements / CondNumElements,
```

Comment:
What if CondNumElements > TrueNumElements?

Reply:
Cannot. `CondNumElements` <= `TrueNumElements`.

Reply:
Add assertion

Reply:
Add in getEntryCost, including comments.

Reply:
Add here too

Reply:
Done.

Reply:
Can it be that `TrueNumElements % CondNumElements != 0`? Will you handle it correctly?

Reply:
Cannot. They are the operands of Instruction::Select. The number of elements must be the same.

Reply:
Add the assertions, please, in all branches

Reply:
Done